packages/typespec-python/src/types.ts

```diff
@@ -216,7 +216,16 @@ function emitProperty<TServiceOperation extends SdkServiceOperation>(
     property: SdkBodyModelPropertyType,
 ): Record<string, any> {
     const isMultipartFileInput = property.multipartOptions?.isFilePart;
-    const sourceType = isMultipartFileInput ? createMultiPartFileType(property.type) : property.type;
+    let sourceType: SdkType | MultiPartFileType = property.type;
+    if (isMultipartFileInput) {
+        sourceType = createMultiPartFileType(property.type);
+    } else if (property.type.kind === "model") {
+            const body = property.type.properties.find((x) => x.kind === "body");
+            if (body) {
+                // for `temperature: HttpPart<{@body body: float64, @header contentType: "text/plain"}>`, the real type is float64
+                sourceType = body.type;
+            }
+    }
```

Comment:
For `temperature: HttpPart<{@body body: float64, @header contentType: "text/plain"}>`, TCGC provide model of `{@body body: float64, @header contentType: "text/plain"}` while python emitter actually only need `float64` so we need to parse it.

Reply:
is it bad that we have `contentType`? I'm thinking we still just pass it through. But I'm good either way

Reply:
`HttpPart` permit us to define more specific info for this part. In this case `@header contentType: "text/plain"` means SDK shall serialize the float64 as plain string instead of json string. Although both actually send same content, JS owner thought we shall add the specific info in the cadl-ranch case then I just added it.

Reply:
I raise an issue https://github.com/Azure/typespec-azure/issues/1488. If all languages agree https://github.com/Azure/typespec-azure/issues/1488#issuecomment-2333048720, TCGC could do the parse logic so that emitters don't need the duplicated work any more.